Sources/Core/Publisher.swift

```diff
@@ -249,4 +255,35 @@ public extension Publisher {
         )
         .eraseToAnyPublisher()
     }
+
+    /// Measure date intervals, starting after some condition is satisfied and ending when another one is satisfied.
+    ///
+    /// - Parameters
+    ///    - measure: A closure called for each output delivered upstream.
+    ///    - startCondition: A closure called to determine when a measurement must start.
+    ///    - endCondition: A closure called to to determine when a measurement must stop.
+    ///
+    /// A measurement is made for each pair of matching conditions.
+    func measureDateInterval(
+        perform measure: @escaping (DateInterval) -> Void,
+        after startCondition: @escaping (Output) -> Bool,
+        when endCondition: @escaping (Output) -> Bool
```

Comment:
```suggestion
        until endCondition: @escaping (Output) -> Bool
```

Reply:
Thanks, I propose `measureEachDateInterval(perform:after:until:)` to make it explicit that measurements always take place between pairs of conditions (and not only the first time).